test/uk/gov/hmrc/tai/service/TaxAccountSummaryServiceSpec.scala

```diff
@@ -0,0 +1,422 @@
+          IncomesSources(livePensionIncomeSources, liveEmploymentIncomeSources, ceasedEmploymentIncomeSources),
+          nonMatchedEmployments)
+      }
+
+
+  }
+
+  def createSUT = new SUT()
+  val taxAccountSummary: TaxAccountSummary = TaxAccountSummary(111, 222, 333.33, 444.44, 111.11)
+  val nonTaxCodeIncome = NonTaxCodeIncome(Some(uk.gov.hmrc.tai.model.domain.income.UntaxedInterest(UntaxedInterestIncome,
+    None, 100, "Untaxed Interest", Seq.empty[BankAccount])), Seq(
+    OtherNonTaxCodeIncome(Profit, None, 100, "Profit")
+  ))
+
+  val liveEmployment1 = TaxCodeIncome(EmploymentIncome, Some(1), 1111, "employment", "1150L", "employer1", OtherBasisOfOperation, Live)
+  val liveEmployment2 = TaxCodeIncome(EmploymentIncome, Some(2), 2222, "employment", "BR", "employer2", Week1Month1BasisOfOperation, Live)
+  val livePension3 = TaxCodeIncome(PensionIncome, Some(3), 3333, "employment", "1150L", "employer3", OtherBasisOfOperation, Live)
+  val livePension4 = TaxCodeIncome(PensionIncome, Some(4), 4444, "employment", "BR", "employer4", Week1Month1BasisOfOperation, Live)
+  val potentiallyCeasedEmployment9 = TaxCodeIncome(EmploymentIncome, Some(9), 1111, "employment", "1150L", "employer9", OtherBasisOfOperation, PotentiallyCeased)
+  val ceasedEmployment10 = TaxCodeIncome(EmploymentIncome, Some(10), 2222, "employment", "BR", "employer10", Week1Month1BasisOfOperation, Ceased)
+  val empEmployment1 = Employment("Employer name1", Some("1ABC"), new LocalDate(2017, 3, 1), None, Seq.empty[AnnualAccount], "DIST1", "PAYE1", 1, None, false, false)
+  val empEmployment2 = Employment("Employer name2", Some("1ABC"), new LocalDate(2017, 3, 1), None, Seq.empty[AnnualAccount], "DIST2", "PAYE2", 2, None, false, false)
+  val pensionEmployment3 = Employment("Pension name1", Some("3ABC"), new LocalDate(2017, 3, 1), None, Seq.empty[AnnualAccount], "DIST3", "PAYE3", 3, None, false, false)
+  val pensionEmployment4 = Employment("Pension name2", Some("4ABC"), new LocalDate(2017, 3, 1), None, Seq.empty[AnnualAccount], "DIST4", "PAYE4", 4, None, false, false)
+  val empEmployment9 = Employment("Employer name3", Some("9ABC"), new LocalDate(2017, 3, 1), None, Seq.empty[AnnualAccount], "DIST9", "PAYE9", 9, None, false, false)
+  val empEmployment10 = Employment("Employer name4", Some("10ABC"), new LocalDate(2017, 3, 1), Some(new LocalDate(2018, 4, 21)), Seq.empty[AnnualAccount], "DIST10", "PAYE10", 10, None, false, false)
+
+
+  val livePensionIncomeSources: Seq[IncomeSource] = Seq(
+    IncomeSource(livePension3, pensionEmployment3),
+    IncomeSource(livePension4, pensionEmployment4)
+  )
+
+  val liveEmploymentIncomeSources: Seq[IncomeSource] = Seq(
+    IncomeSource(liveEmployment1, empEmployment1),
+    IncomeSource(liveEmployment2, empEmployment2)
+  )
+
+  val ceasedEmploymentIncomeSources: Seq[IncomeSource] = Seq(
+    IncomeSource(potentiallyCeasedEmployment9, empEmployment9),
+    IncomeSource(ceasedEmployment10, empEmployment10)
+  )
```

Comment:
This test set up looks the same as the one in the controller, I am assuming you can delete all the setup in the controller and keep these ones.

Reply:
Have took all of the test data setup and moved into a shared trait